### PR TITLE
FORMS-1327: Add proxy, externalAPI and adminExternalAPI to open api spec

### PR DESCRIPTION
**Pull request by usingtechnology**
# Description
Updating the Open API spec to include apis related to External API.
These include the calls for :
- managing the External API for a form
- CHEFS administration functions for approving External APIs
- generating the proxy headers and calling the proxy (external) api.
- 

## Type of Change

docs (change to documentation

## Checklist

- [x] I have read the [CONTRIBUTING](/bcgov/common-hosted-form-service/blob/main/CONTRIBUTING.md) doc
- [ ] I have checked that unit tests pass locally with my changes
- [ ] I have run the npm script lint on the frontend and backend
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have approval from the product owner for the contribution in this pull request

### CODE REVIEW

File: app/src/docs/v1.api-spec.yaml [hunk 1]
```diff
@@ -3663,6 +4076,47 @@ components:
...
+          description: Ministry Name of Form
+          example: CITZ
+        formName:
+          type: string
+          description: Name of the ExternalApi - unique per form.
+          example: My Form
+        endpointUrl:
+          type: string
+          format: url
+          description: Name of the ExternalApi - unique per form.
```
Comment (RyanBirtch-aot) [inline app/src/docs/v1.api-spec.yaml:4107]: Maybe change to "URL endpoint used for the external api"?

File: app/src/docs/v1.api-spec.yaml [hunk 2]
```diff
@@ -3830,6 +4284,57 @@ components:
...
+          description: Form ID this ExternalApi belongs to.
+          example: 1223h78b-21d2-4128-b4e8-02ad3daipoi32
+        name:
+          type: string
+          description: Name of the ExternalApi - unique per form.
+          example: BC Registries Number Lookup
+        endpointUrl:
+          type: string
+          format: url
+          description: Name of the ExternalApi - unique per form.
```
Comment (RyanBirtch-aot) [inline app/src/docs/v1.api-spec.yaml:4307]: Maybe change to "URL endpoint used for the external api"?

File: app/src/docs/v1.api-spec.yaml (hunk 1 above)
Comment (usingtechnology) [reply app/src/docs/v1.api-spec.yaml]: Good catch, thanks. Fixing.

File: app/src/docs/v1.api-spec.yaml (hunk 2 above)
Comment (usingtechnology) [reply app/src/docs/v1.api-spec.yaml]: See above. Thanks.

**Review by RyanBirtch-aot — APPROVED** [review]

**Comment by WalterMoar** [comment]
Reading the deployed documentation might help prevent errors due to formatting (and make the reading a little easier).

**Pull request merged**